File: C# Fundamentals/C# OOP Basics/13. Polymorphism/Ex/P01_Vehicles/Models/Truck.cs
using System;

public class Truck : Vehicle
{
    public Truck(double fuelQuant, double fuelConsum, double tankCapacity) : base(fuelQuant, fuelConsum, tankCapacity)
    {
        this.FuelConsumptionPerKm = fuelConsum + TruckConsumptionIncrease;
    }

    public override void Refuel(double liters)
    {
        if (liters <= 0)
        {
            throw new InvalidOperationException(FuelMustBePositiveMessage);
        }
        if (this.FuelQuantity + liters > this.FuelTankCapacity)
        {
            throw new InvalidOperationException(string.Format(FuelIsMoreThanCapacityMessage, liters));
        }
        else
        {
            this.FuelQuantity += TruckFuelLeak * liters;
        }
        
    }
}

File: Java/01.Advanced/02.OOP/11.Polymorphism/Ex/src/t_02_vehicles_extend/Truck.java
package t_02_vehicles_extend;

public class Truck extends Vehicle {
    public Truck(double fuelQuantity, double fuelConsumption, double tankCapacity) {
        super(fuelQuantity, fuelConsumption, tankCapacity);
        this.setAddedFuelConsumption(1.6);
    }

    @Override
    public void refuel(double fuel) {
        super.refuel(fuel  * 0.95);
    }
}
